File: bindings/rust/benches/bench_message.rs
//! EncryptMessage throughput vs plaintext size (Single Message
//! profile).

use std::time::Duration;

use criterion::{Criterion, Throughput, criterion_group, criterion_main};
use itb::{OptsBuilder, Pipeline};

fn bench_message(c: &mut Criterion) {
    let pipe = Pipeline::init("singlemsg-triple-mac-v1", &OptsBuilder::new()).unwrap();
    let mut group = c.benchmark_group("encrypt_message");
    group
        .sample_size(10)
        .measurement_time(Duration::from_secs(2));
    for size in [1usize << 10, 1 << 16, 1 << 20, 16 << 20] {
        let plain = vec![0xA5u8; size];
        group.throughput(Throughput::Bytes(size as u64));
        group.bench_function(format!("{size}B"), |b| {
            b.iter(|| pipe.encrypt_message(&plain).unwrap());
        });
    }
    group.finish();
}

criterion_group!(benches, bench_message);
criterion_main!(benches);

File: bindings/swift/Sources/ItbBench/bench_message.swift
/*
 * bench_message.swift — encryptMessage throughput vs plaintext size
 * (Single Message profile) at 1 MiB / 16 MiB / 64 MiB.
 */

import Foundation
import Itb

func runMessageBench() {
    do {
        let opts = try benchBuildOpts()
        let pipe = try Pipeline(profile: benchProfileName("singlemsg-triple-nomac-v1"),
                                opts: opts)
        for size in [1 << 20, 16 << 20, 64 << 20] {
            let plain = benchPayload(size)
            benchCase("message", size) {
                _ = try pipe.encryptMessage(plain)
            }
        }
    } catch {
        FileHandle.standardError.write(
            Data("bench_message: init failed: \(error)\n".utf8))
        exit(1)
    }
}
